Make DZ2 Input Boxes

MarcoPolo opened the pull request:
@keybase/react-hackers 

Matches the new designs for inputs. Wasn't too bad, I mostly made it a flex box. Just had to undo some styling from material ui and a couple tweaks.

Multiline is super hacky on the material-ui side, so there's a couple hacks to get that working properly with the styles.

## Todo: 

- [x] Feature flag it

chrisnojima commented:
I think this is using noto sans and not lato?

chrisnojima commented:
Floating label is supposed to be light blue

chrisnojima commented:
Might be cool to show the example of the error text being pushed down like it is in the style guide (setting errorStyle marginTop:20 or whatever it is

MarcoPolo commented:
> I think this is using noto sans and not lato?
The other PR does the font stuff.

MarcoPolo commented:
This won't need a feature flag, because the only place you can see the input component is the menubar, and that doesn't look any different

chrisnojima commented:
we have dz2 designs for pinentry also

chrisnojima commented:
one case we shouldl make sure works is the mix of hint text and floating text. when i wrote it i had small override stuff but i think we should remove that (or allow that to be overridden). In the paperkey page (https://app.zeplin.io/project.html#pid=56b25ffc64fc91773a73aaf8&sid=56c33234997a3c1216ff4378) the hint text is a multiline example of a paperkey, we definitely don't want that to be floating when they focus

MarcoPolo commented:
Updated!
Added a new component for forms with checkboxes under them. Preview:
<img width="272" alt="screen shot 2016-02-16 at 2 29 30 pm" src="https://cloud.githubusercontent.com/assets/594035/13093388/d9461870-d4b9-11e5-8c91-ea3988e78234.png">

Also supports multiline hint texts.

Feature flags itself by calling the old component, when we move away we can just delete that whole file.

Not pixel perfect for the form checkbox stuff, but I'd rather get that in another PR.

chrisnojima commented on shared/common-adapters/input.desktop.js:
```diff
@@ -1,34 +1,42 @@
 export default class Input extends Component {
-  constructor (props) {
+  props: Props;
+  state: {value: ?string, focused: boolean};
```
can't we put all these annotations into input.js.flow?

chrisnojima commented:
:+1: % .js.flow comment
